### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by ssaviak**
[DEMO LINK](https://ssaviak.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by yaneznayou — APPROVED** [review]
Looks good

**Review by olya-shyian — CHANGES_REQUESTED** [review]
I love it! Very easy to read! Great work🤩

File: src/components/TodoModal/TodoModal.tsx [hunk 1]
```diff
@@ -1,49 +1,79 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
```
Comment (DanilWeda) [inline src/components/TodoModal/TodoModal.tsx:8]: use interface prefer

File: src/components/TodoFilter/TodoFilter.tsx [hunk 2]
```diff
@@ -1,34 +1,78 @@
...
+          onChange={(e) => setQuery(e.target.value)}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+        {query && (
```
Comment (DanilWeda) [inline src/components/TodoFilter/TodoFilter.tsx:64]:
```suggestion
        {!!query && (
```
in rendering with && for best rendering control use !! to return a boolean value

**Review by DanilWeda — APPROVED** [review]
Great! 
Approved. 
Many thanks!

**Pull request closed without merging**